### PR TITLE
Rename attachments props to images props

### PR DESCRIPTION
**Pull request by dtrucs**
Images and documents are provided by the API under an "attachments" key.

In GTR, only images are displayed, filtered by the key `type="image"`.
This property should be called "images", but it's actually "attachments". However, in a future development, "document" attachments will also be displayed, making the "attachments" key ambiguous.

To avoid this ambiguity, a renaming is necessary.

### CODE REVIEW

**Review by camillemonchicourt — APPROVED** [review]

File: frontend/src/components/ImageWithLegend/index.tsx [hunk 1]
```diff
@@ -25,7 +25,7 @@ export const ImageWithLegend: React.FC<ImageWithLegendProps> = ({
 }) => {
   const figureId = useId();
   const imageId = useId();
-  const legend = [attachment?.legend, attachment?.author].filter(Boolean).join(' - ');
+  const legend = [image?.legend, image?.author].filter(Boolean).join(' - ');
```
Comment (mabhub) [inline frontend/src/components/ImageWithLegend/index.tsx:28]:
```suggestion
  const legend = [image.legend, image.author].filter(Boolean).join(' - ');
```

File: frontend/src/components/ImageWithLegend/index.tsx [hunk 2]
```diff
@@ -34,11 +34,11 @@ export const ImageWithLegend: React.FC<ImageWithLegendProps> = ({
       className={`relative ${className}`}
     >
       <img
-        alt={attachment?.legend ?? ''}
+        alt={image?.legend ?? ''}
         className={cn(`object-cover object-center overflow-hidden size-full ${classNameImage}`)}
         id={imageId}
         loading={loading ?? 'eager'}
-        src={attachment?.url ? attachment.url : getGlobalConfig().fallbackImageUri}
+        src={image?.url ? image.url : getGlobalConfig().fallbackImageUri}
```
Comment (mabhub) [inline frontend/src/components/ImageWithLegend/index.tsx:41]:
```suggestion
        src={image.url || getGlobalConfig().fallbackImageUri}
```

```suggestion
        src={image.url || getGlobalConfig()?.fallbackImageUri}
```

**Review by mabhub — CHANGES_REQUESTED** [review]

File: frontend/src/components/pages/details/components/DetailsCard/DetailsCard.tsx [hunk 3]
```diff
@@ -35,7 +35,7 @@ export const DetailsCard: React.FC<DetailsCardProps> = ({
   id,
   name,
   description,
-  attachments,
+  images,
```
[the remark this answers is not in the document]
Comment (mabhub) [reply frontend/src/components/pages/details/components/DetailsCard/DetailsCard.tsx:38]: Maybe at app level, but not at the component level.
If `DetailsCard` is instanced without props, an exception will be thrown.

[the remark this answers is not in the document]
Comment (dtrucs) [reply frontend/src/components/pages/details/components/DetailsCard/DetailsCard.tsx:38]: TypeScript won't let you do this! :sweat_smile:

[the remark this answers is not in the document]
Comment (mabhub) [reply frontend/src/components/pages/details/components/DetailsCard/DetailsCard.tsx:38]: TypeScript is not present at runtime.  
Props may be conditionally provided based on data from external API.

**Review by mabhub — APPROVED** [review]

**Pull request merged**